### PR TITLE
Updated kubevirtci prometheus stack to more recent versions

### LINKED ISSUE #1154: monitoring: Update kubevirtci prometheus stack to more recent versions
The monitoring stack that is deployed as part of kubevirtci has not been updated in some time. 

This stack includes a number of services[1] and is based on [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) :
- alertmanager
- grafana
- node-exporter
- prometheus
- prometheus operator

The manifests for each of these services will have to updated.

Follow the KUBEVIRTCI_LOCAL_TESTING.md document to test the updates locally. 
https://github.com/kubevirt/kubevirtci/blob/main/KUBEVIRTCI_LOCAL_TESTING.md#start-cluster-with-prometheus-alertmanager-and-grafana

As part of this task it may be good to capture the steps taken as part of the upgrade as the document is lacking details:
https://github.com/kubevirt/kubevirtci/tree/main/cluster-provision/k8s/1.29/manifests/prometheus#upgrading

[1] https://github.com/kubevirt/kubevirtci/tree/main/cluster-provision/k8s/1.29/manifests/prometheus

### PR DESCRIPTION
**Pull request by anishbista60**
**What this PR does / why we need it**:
To update  the prometheus stack to recent version

**Which issue(s) this PR fixes** *(optional, in `fixes #1154(, fixes #1154, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1154

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by anishbista60** [comment]
@brianmcarey   as of now , i have only changed the manifest for `alertmanager` . The confusion is   do i need to change the entire manifest or only version. 
Like 
```
  -  app.kubernetes.io/version: 0.21.0
  + app.kubernetes.io/version: 0.27.0
```

please suggest i will do it accordingly . if the changes are good then  only i will go for other 
- grafana
- node-exporter
- prometheus
- prometheus operator

File: cluster-provision/k8s/1.31/manifests/prometheus/alertmanager/alertmanager-alertmanager.yaml [hunk 1]
```diff
@@ -2,34 +2,36 @@ apiVersion: monitoring.coreos.com/v1
...
   podMetadata:
     labels:
       app.kubernetes.io/component: alert-router
+      app.kubernetes.io/instance: main
       app.kubernetes.io/name: alertmanager
       app.kubernetes.io/part-of: kube-prometheus
-      app.kubernetes.io/version: 0.21.0
-  replicas: 1
+      app.kubernetes.io/version: 0.27.0
+  replicas: 3
```
Comment (brianmcarey) [inline cluster-provision/k8s/1.31/manifests/prometheus/alertmanager/alertmanager-alertmanager.yaml:23]: We won't need three replicas in our test clusters - its better to keep this at 1.

**Review by brianmcarey — COMMENTED** [review]
Changes look good for alertmanager just one comment on the replicas

File: cluster-provision/k8s/1.31/manifests/prometheus/alertmanager/alertmanager-alertmanager.yaml (hunk 1 above)
Comment (anishbista60) [reply cluster-provision/k8s/1.31/manifests/prometheus/alertmanager/alertmanager-alertmanager.yaml:23]: @brianmcarey ok then i will modify in all folder . Thankyou

Comment (anishbista60) [reply cluster-provision/k8s/1.31/manifests/prometheus/alertmanager/alertmanager-alertmanager.yaml]: @brianmcarey i have modified to all the manifest to the recent version. Please check .

**Comment by anishbista60** [comment]
@brianmcarey Could you please review the changes ? for me looks good.

**Comment by anishbista60** [comment]
@brianmcarey i updated the tag but still not working . 
```
E0723 08:04:20.685263    3467 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = reading manifest 11.1.0 in quay.io/kubevirtci/grafana-grafana: manifest unknown" image="quay.io/kubevirtci/grafana-grafana:11.1.0"
```

File: cluster-provision/k8s/1.31/manifests/prometheus/prometheus/prometheus-prometheus.yaml [hunk 2]
```diff
@@ -16,34 +16,35 @@ spec:
...
       app.kubernetes.io/name: prometheus
       app.kubernetes.io/part-of: kube-prometheus
-      app.kubernetes.io/version: 2.26.0
+      app.kubernetes.io/version: 2.53.0
   podMonitorNamespaceSelector: {}
   podMonitorSelector: {}
   probeNamespaceSelector: {}
   probeSelector: {}
-  replicas: 1
+  replicas: 2
```
Comment (brianmcarey) [inline cluster-provision/k8s/1.31/manifests/prometheus/prometheus/prometheus-prometheus.yaml:35]: I think we just need one replica here as this is just a test cluster

Comment (anishbista60) [reply cluster-provision/k8s/1.31/manifests/prometheus/prometheus/prometheus-prometheus.yaml]: @brianmcarey done

**Comment by brianmcarey** [comment]
> @brianmcarey Now, no idea why ci is faling?

It looks like the grafana pod is failing to reach a running state - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/1228/check-provision-k8s-1.31/1815785151704076288#1:build-log.txt%3A3055

**Comment by anishbista60** [comment]
@brianmcarey please test it ?

**Comment by anishbista60** [comment]
@brianmcarey stuck in container creating state . could you please help me  ?

**Comment by brianmcarey** [comment]
> @brianmcarey Again grafana pod is stuck in container creating state . could you please help me ?

Yes sure I can try to take a look tomorrow. Will let you know if I see anything.

File: cluster-provision/k8s/1.31/manifests/prometheus/grafana/grafana-deployment.yaml [hunk 3]
```diff
@@ -201,9 +243,9 @@ spec:
       - configMap:
           name: grafana-dashboard-scheduler
         name: grafana-dashboard-scheduler
-      - configMap:
-          name: grafana-dashboard-statefulset
-        name: grafana-dashboard-statefulset
       - configMap:
           name: grafana-dashboard-workload-total
         name: grafana-dashboard-workload-total
+      - name: grafana-config
```
Comment (brianmcarey) [inline cluster-provision/k8s/1.31/manifests/prometheus/grafana/grafana-deployment.yaml:249]: it looks like you are missing this secret 

```
./cluster-up/kubectl.sh describe pod -n monitoring grafana-7f998f499d-cs58h | grep Warning
selecting podman as container runtime
  Warning  FailedMount  79s (x9 over 3m27s)  kubelet            MountVolume.SetUp failed for volume "grafana-config" : secret "grafana-config" not found
```

Comment (anishbista60) [reply cluster-provision/k8s/1.31/manifests/prometheus/grafana/grafana-deployment.yaml:249]: @brianmcarey  Thankyou

**Comment by anishbista60** [comment]
@brianmcarey i think calling you everytime for testing may overhelmed you . So , i made entire unit testing setup in my local machine 

```
8f267f3271c4   quay.io/kubevirtci/k8s-1.30:2408071207-c848512b   "/bin/bash -c '/vm.s…"   45 minutes ago   Up 45 minutes                                                                                                                                                                                                                                                                                                                                                                                                                                                                               k8s-1.30-node01
246fa1ac43d3   quay.io/libpod/registry:2.7                       "/entrypoint.sh /etc…"   45 minutes ago   Up 45 minutes                                                                                                                                                                                                                                                                                                                                                                                                                                                                               k8s-1.30-registry
cf8da9433a97   quay.io/kubevirtci/k8s-1.30:2408071207-c848512b   "/bin/bash -c /dnsma…"   45 minutes ago   Up 45 minutes   127.0.0.1:8443->8443/tcp, 127.0.0.1:31111->31111/udp, 0.0.0.0:32776->80/tcp, :::32776->80/tcp, 0.0.0.0:32775->443/tcp, :::32775->443/tcp, 0.0.0.0:32774->2201/tcp, :::32774->2201/tcp, 0.0.0.0:32773->5000/tcp, :::32773->5000/tcp, 0.0.0.0:32772->5901/tcp, :::32772->5901/tcp, 0.0.0.0:32771->6443/tcp, :::32771->6443/tcp, 0.0.0.0:32770->30007/tcp, :::32770->30007/tcp, 0.0.0.0:32769->30008/tcp, :::32769->30008/tcp, 0.0.0.0:32768->31001/tcp, :::32768->31001/tcp   k8s-1.30-dnsmasq
126394a153a8   52edc51989ac                                      "/bin/sh -c '/downlo…"   3 hours ago      Up 3 hours    
```

Now, how to see the logs of pod as you shared before . 
```
./cluster-up/kubectl.sh describe pod -n monitoring grafana-7f998f499d-cs58h | grep Warning
selecting podman as container runtime
  Warning  FailedMount  79s (x9 over 3m27s)  kubelet            MountVolume.SetUp failed for volume "grafana-config" : secret "grafana-config" not found
```

when i tried running `.cluster-up/kubeclt.sh get pod -A `  getting error 

```
FATAL: either KUBEVIRTCI_TAG or KUBEVIRTCI_GOCLI_CONTAINER must be set
```
and i have export this `ENV` also still not working

**Comment by anishbista60** [comment]
@brianmcarey SORRY for bothoring you alot. i have provisioned local cluster in my local machine for testing. Now , how to test that `code` . Asking for installtion of crd when i try to deployed the grafana pod.

**Comment by brianmcarey** [comment]
> @brianmcarey SORRY for bothoring you alot. i have provisioned local cluster in my local machine for testing. Now , how to test that `code` . Asking for installtion of crd when i try to deployed the grafana pod.

For testing changes locally - this doc is pretty useful. https://github.com/kubevirt/kubevirtci/blob/main/KUBEVIRTCI_LOCAL_TESTING.md

 Grafana and Prometheus should be installed on cluster up but you will need the following env variables set
```
export KUBEVIRT_DEPLOY_PROMETHEUS=true
export KUBEVIRT_DEPLOY_GRAFANA=true
```

**Comment by anishbista60** [comment]
@brianmcarey  should i set it to `1.31 ` or leave as it is.
```
export KUBEVIRT_PROVIDER=k8s-1.30
```

**Comment by brianmcarey** [comment]
> @brianmcarey should i set it to `1.31 ` or leave as it is.
> 
> ```
> export KUBEVIRT_PROVIDER=k8s-1.30
> ```



> @brianmcarey should i set it to `1.31 ` or leave as it is.
> 
> ```
> export KUBEVIRT_PROVIDER=k8s-1.30
> ```
> 
> my output look like and not getting the grafana , promethus even after exporting above `env` .
> 
> ```
> ./cluster-up/kubectl.sh get pod -A
> selecting docker as container runtime
> NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE
> default       local-volume-provisioner-9lcns             1/1     Running   0          3m12s
> kube-system   calico-kube-controllers-589c88b6d7-gsnwr   1/1     Running   0          3m12s
> kube-system   calico-node-dd9ws                          1/1     Running   0          3m12s
> kube-system   coredns-98fcf86f9-87l97                    1/1     Running   0          3m12s
> kube-system   coredns-98fcf86f9-zm6c8                    1/1     Running   0          3m12s
> kube-system   etcd-node01                                1/1     Running   1          3m27s
> kube-system   kube-apiserver-node01                      1/1     Running   1          3m27s
> kube-system   kube-controller-manager-node01             1/1     Running   1          3m27s
> kube-system   kube-proxy-x8xdg                           1/1     Running   0          3m12s
> kube-system   kube-scheduler-node01                      1/1     Running   1          3m27s
> ```

You should set it to
```
export KUBEVIRT_PROVIDER=k8s-1.31
```

**Comment by anishbista60** [comment]
@brianmcarey  Even after creating secret `grafana-config`.  it is not deploying in cluster automatically . Could you please test it. 

```
k get secret -n monitoring
NAME                             TYPE     DATA   AGE
alertmanager-main                Opaque   1      18m
alertmanager-main-generated      Opaque   1      18m
alertmanager-main-tls-assets-0   Opaque   0      18m
alertmanager-main-web-config     Opaque   1      18m
grafana-datasources              Opaque   1      18m
prometheus-k8s                   Opaque   1      18m
prometheus-k8s-tls-assets-0      Opaque   0      18m
prometheus-k8s-web-config        Opaque   1      18m
```

**Comment by brianmcarey** [comment]
> @brianmcarey Even after creating secret `grafana-config`. it is not deploying in cluster automatically . Could you please test it.
> 
> ```
> k get secret -n monitoring
> NAME                             TYPE     DATA   AGE
> alertmanager-main                Opaque   1      18m
> alertmanager-main-generated      Opaque   1      18m
> alertmanager-main-tls-assets-0   Opaque   0      18m
> alertmanager-main-web-config     Opaque   1      18m
> grafana-datasources              Opaque   1      18m
> prometheus-k8s                   Opaque   1      18m
> prometheus-k8s-tls-assets-0      Opaque   0      18m
> prometheus-k8s-web-config        Opaque   1      18m
> ```

I think you probably need to add it here  - https://github.com/kubevirt/kubevirtci/blob/c848512b0622a690cd61d6e65044d4b1fa91cff8/cluster-provision/k8s/1.31/prometheus.sh#L116

**Comment by brianmcarey** [comment]
@anishbista60 - nearly there - just to update a couple of images in the prepull list  - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/1228/check-provision-k8s-1.31/1821543575784001536#1:build-log.txt%3A3121

**Comment by anishbista60** [comment]
@brianmcarey done

**Comment by anishbista60** [comment]
@brianmcarey looks fine . what next ?

**Review by brianmcarey — COMMENTED** [review]
Looks good @anishbista60 - just one little nit comment.

File: cluster-provision/k8s/1.31/extra-pre-pull-images [hunk 4]
```diff
@@ -8,3 +8,5 @@ quay.io/calico/cni:v3.18.0
 quay.io/calico/kube-controllers:v3.18.0
 quay.io/calico/node:v3.18.0
 quay.io/calico/pod2daemon-flexvol:v3.18.0
+quay.io/kubevirtci/grafana-grafana:11.1.0
```
[the remark this answers is not in the document]
Comment (anishbista60) [reply cluster-provision/k8s/1.31/extra-pre-pull-images:10]: @brianmcarey done

**Comment by anishbista60** [comment]
@brianmcarey Is there anything pending from my side?

File: cluster-provision/k8s/1.31/extra-pre-pull-images [hunk 5]
```diff
@@ -8,3 +7,5 @@ quay.io/calico/cni:v3.18.0
...
+quay.io/prometheus-operator/prometheus-config-reloader:v0.75.1
```
Comment (brianmcarey) [inline cluster-provision/k8s/1.31/extra-pre-pull-images:11]: It looks like there is an older version of this image - https://github.com/kubevirt/kubevirtci/pull/1228/files#diff-f4afa2d47e15ef77f00b37aa0b6308240dd8923fd1a933c12dacf09f7bf48b17R5

**Review by brianmcarey — COMMENTED** [review]
This is probably worth applying to the other providers too - could you copy these changes to k8s-1.28, k8s-1.29 & k8s-1.30.

We should be good to merge after that.

File: cluster-provision/k8s/1.31/extra-pre-pull-images (hunk 5 above)
Comment (anishbista60) [reply cluster-provision/k8s/1.31/extra-pre-pull-images:11]: @brianmcarey should i create the different commit for each provider  or append all the changes to the present coommit

Comment (brianmcarey) [reply cluster-provision/k8s/1.31/extra-pre-pull-images:11]: Different commits for each provider would be good.

Comment (anishbista60) [reply cluster-provision/k8s/1.31/extra-pre-pull-images:11]: @brianmcarey thank you

**Comment by anishbista60** [comment]
@brianmcarey updated everything . could you please go for testing ?

**Comment by brianmcarey** [comment]
/test all 

Thanks @anishbista60  - can you just update the commit messages so that they make sense standalone? A commit message of "k8s-1.28" does not explain the change well.

**Comment by anishbista60** [comment]
@brianmcarey Everything done.  looks fine . Please test it once. Yesterday, all test cases were passed except `check-up-kind-1.27-vgpu	`.

**Review by brianmcarey — APPROVED** [review]
/approve

**Comment by anishbista60** [comment]
@brianmcarey Thankyou for assisting me in this PR.

**Pull request merged**